DOC-1138/vop-001-vop-mvp-documentation (#382)
swan-io/docs: docs/preview/vop/guide-verify-beneficiary.mdx

      }
    }
  ) {
    ... on InitiateCreditTransfersSuccessPayload {
      __typename
      beneficiaryVerificationResults {
        status
        ... on VerifyBeneficiaryMatch {
          __typename
          status
        }
        ... on VerifyBeneficiaryCloseMatch {
          __typename
          status
          nameSuggestion
        }
        ... on VerifyBeneficiaryNoMatch {
          __typename
          status
        }
        ... on VerifyBeneficiaryNotPossible {
          __typename
          status
        }
      }
      payment {
        statusInfo {
          ... on PaymentConsentPending {
            __typename
            consent {
              consentUrl
            }
          }
        }
      }
    }
    }
    ... on BeneficiaryVerificationTokenAlreadyConsumedRejection {
      __typename
    }
    ... on InternalErrorRejection {
      __typename
    }
    ... on ForbiddenRejection {
      __typename
    }
    ... on AccountNotFoundRejection {
      id
    }
  }
}
```

The `InitiateCreditTransfersSuccessPayload` includes:

> - **`beneficiaryVerificationResults`**: Array of `VerifyBeneficiaryResult` ordered by result types.

Author: Array is the technical term for this type of object, usually denoted by `[item1, item2]` notation

> The `InitiateCreditTransfersSuccessPayload` includes:
> - **`beneficiaryVerificationResults`**: Array of `VerifyBeneficiaryResult` ordered by result types.
> - Standard payment information.

Author: ```suggestion
The `InitiateCreditTransfersSuccessPayload` includes the `beneficiaryVerificationResults`, an array of `VerifyBeneficiaryResult` objects ordered by result types.
```

> ```json title="Returned Payload" {7,9,16}

Author: ```suggestion
```json title="Payload" {7,9,16}
```

          }
        }
      }
    }
  }
}
```

## Token validation {#token-validation}

**Token requirements**:
- Tokens are validated for authenticity, status, and expiration.
- Invalid, expired, or undecryptable tokens result in a `ValidationRejection`.
- A token is consumed after it has been used in one initiation. Already consumed tokens result in a `BeneficiaryVerificationTokenAlreadyConsumedRejection`.
- Tokens expire 24 hours after generation.
:::tip
Note that the same token can be used multiple times in the same initiation (meaning multiple credit transfers to the same beneficiary, within the same API call).
:::

## Bulk payments {#bulk-payments}

For bulk credit transfers, each transfer in the batch **must** include a valid `beneficiaryVerificationToken` obtained through the `verifyBeneficiary` mutation. If any transfer in a batch is submitted with full beneficiary details instead of a token, the entire batch initiation will be rejected with a `ForbiddenRejection`.

### Company opt-out for bulk transfers {#company-opt-out}

Company end-users can opt-out of VoP for bulk transfers through the `updateAccountSettings` mutation. Individual end-users cannot opt-out.

> :::info

Author: We will update it next week and it would cause a linebreak in the right sidebar, so I'll hold off for now :)

    }
  ) {
    ... on UpdateAccountSettingsSuccess {
      account {
        settings {
          allowBulkCreditTransfersWithoutBeneficiaryVerification
        }
      }
    }
    ... on AccountNotFoundRejection {
      message
    }
    ... on ForbiddenRejection {
      message
    }
    ... on AccountHolderNotFoundRejection {
      message
    }
  }
}
```

**Requirements**:
- Only available for Company account types.
- `consentRedirectUrl` required for the settings change flow.
-->

## Sandbox testing {#sandbox-testing}

In the Sandbox environment, the `verifyBeneficiary` mutation returns simulated verification results based on the beneficiary name pattern:

| Name starts with | Returns result |
|------------------|----------------|
| "Match" | `Match` |
| "CloseMatch" | `CloseMatch` |
| "NoMatch" | `NoMatch` |
| "NotPossible" | `VerificationNotPossible` |

**Example**: Beneficiary name "Match Henri Dupont" returns a `Match` result.

:::tip
Note that if a trusted beneficiary ID is used as input, the verification will be based on the Trusted Beneficiary name.
:::

### Testing bulk payments {#testing-bulk-payments}

<FieldRequirementsLegend />

> By default, in Sandbox:

Author: ```suggestion
#### Default sandbox requirements
```
